Libplanet.Net/Consensus/ConsensusContext.cs

```diff
@@ -351,14 +388,15 @@ private void ClearOldLastCommitCache(long maxSize = 30)
         /// <param name="height">The upper bound of height of the contexts to be discarded.</param>
         private void RemoveOldContexts(long height)
         {
-            var values = _contexts.Values;
-            foreach (var ctx in values)
+            lock (_contextLock)
             {
-                if (ctx.Height < height)
+                var values = _contexts.Values;
+                foreach (var ctx in values)
```

Comment:
```csharp
                foreach (var ctx in _contexts.Values)
```

We can do away with extraneous variable `values` since `_contexts` is locked now.